internal/module/helper.go

```diff
@@ -37,14 +30,14 @@ func IntermediateImageName(name, namespace, targetImage string) string {
 
 // ShouldBeBuilt indicates whether the specified KernelMapping of the
 // Module should be built or not.
-func ShouldBeBuilt(modSpec kmmv1beta1.ModuleSpec, km kmmv1beta1.KernelMapping) bool {
-	return modSpec.ModuleLoader.Container.Build != nil || km.Build != nil
+func ShouldBeBuilt(km kmmv1beta1.KernelMapping) bool {
+	return km.Build != nil
 }
 
 // ShouldBeSigned indicates whether the specified KernelMapping of the
 // Module should be signed or not.
-func ShouldBeSigned(modSpec kmmv1beta1.ModuleSpec, km kmmv1beta1.KernelMapping) bool {
-	return modSpec.ModuleLoader.Container.Sign != nil || km.Sign != nil
+func ShouldBeSigned(km kmmv1beta1.KernelMapping) bool {
```

Comment:
If we agree that a `Module`'s sign section should be used as default when it's not defined in the kernel mapping, then should this function accept a [`*Sign`](https://github.com/kubernetes-sigs/kernel-module-management/blob/main/api/v1beta1/module_types.go#L70) instead?

Reply:
By the time we call this function ( i assume you are referring to ShouldBeSigned), KernelMapping should contain all unified configuration. So, in case origiinal KM sign section was not defined, the Sign section in the passed kernel mapping will contain the Module's sing section.  We can change the function to receive only Sign section, i just thought it would be more clear API that the decision is made based on the data in KernelMapping